### ISSUE TITLE
How to run experiment with only photos?

### ISSUE THREAD

**Issue by franciscoWizz** [POSTECH-CVLab/SCNeRF#3]
Hi Mr,

I would like to run an experiment with your model using a list of pictures from an object to get the estimated camera poses of each picture. How can I mount that experiment?

Thanks in advance,

**Comment by jeongyw12382** [comment]
Sorry for being late. I have added descriptions on the new issue you have just added. 
@xufengfan96

**Comment by jeongyw12382** [comment]
Please reopen the issue whenever you need help in this issue.

**Comment by hmdolatabadi** [comment]
@jeongyw12382 Hi. Thanks for the interesting paper and providing the code. I read through the custom code run snippet that you provided above:

`python run_nerf.py --config configs/llff_data/flower.txt --expname $(basename "${0%.*}") --chunk 8192 --N_rand 1024 --camera_model pinhole_rot_noise_10k_rayo_rayd --ray_loss_type proj_ray_dist --multiplicative_noise True --i_ray_dist_loss 10 --grid_size 10 --ray_dist_loss_weight 0.0001 --N_iters 800001 --ray_o_noise_scale 1e-3 --ray_d_noise_scale 1e-3 --add_ie 200000 --add_od 400000 --add_prd 600000 --lrate_decay 400 --dataset_type custom --run_without_colmap both
`

Is this going to run NeRF or SCNeRF? Cause I saw a little difference with the .sh files in the original repo.
Also, could you please tell that after training is done, how we can render a video using our trained model?

Thanks for your help in advance.

**Comment by vishnukool** [comment]
Hi @jeongyw12382 
Two more questions, if you don't mind: 

1. Does the code in "custom" branch with `--dataset_type custom` work for 360 degree scenes like the "tanks_and_temples" images ? Or is it only for forward facing scenes like LLFF fern dataset?
2. If it does work for 360 scenes, can you confirm that it doesn't need any COLMAP camera parameters, initialization, etc. ?

**Comment by jeongyw12382** [comment]
@hmdolatabadi 
The script you've mentioned will run SCNeRF since the camera model is set to "pinhole_rot_noise_10k_rayo_rayd." --ft_path loads the pre-trained model. 
Thus, if you are running four stages independently, then you should add the ft_path to load the pre-trained model of the previous stage. However, if you run the script above, the code will automatically run the four stages sequentially.  If you need more help, please let me know.

**Comment by jeongyw12382** [comment]
@vishnukool 
I'll respond to this issue on your newly uploaded issue.

**Comment by hmdolatabadi** [comment]
@jeongyw12382 Thanks a lot. I will checkout the links and functions and see how it goes. Thanks again.

**Comment by hmdolatabadi** [comment]
@jeongyw12382 I tried training a model on images only, and everything went well. Now, to generate a video sequence, I tried the above codes, but all of them need the `render_poses` variable. I guess my original question comes back to this: when you only have images, how do you generate the `render_poses` variable so that you can generate/render images? Thanks.

**Comment by jeongyw12382** [comment]
It depends on the type of camera trajectories you want to render. 

1. If you want to generate a 360-scene video, then based on the estimated pose, you can generate the spherical pose with the code here. 
https://github.com/POSTECH-CVLab/SCNeRF/blob/dc57e9f6e763284a12bed4812e6945f49ee0ef5e/NeRF/load_blender.py#L144 

2. If you want to generate the same camera trajectory with the train/val/test set (assuming video), then there are two choices
- If frames are sufficiently many, then you can directly re-render the train/val/test set to generate the video. 
- If frames are insufficiently many, then you should implement codes that estimate intermediate camera trajectory from given frames(poses).

3. If you want to generate a circular path (refer to the LLFF dataset video), then you should utilize the code below.
https://github.com/POSTECH-CVLab/SCNeRF/blob/dc57e9f6e763284a12bed4812e6945f49ee0ef5e/NeRF/load_llff.py#L299

4. Otherwise, probably you should implement codes generating a camera trajectory.
